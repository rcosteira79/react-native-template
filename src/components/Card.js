import React from 'react';
import { View } from 'react-native';

const Card = props => {
  return <View style={[styles.containerStyle, props.style]}>{props.children}</View>;
};

const styles = {
  containerStyle: {
    borderWidth: 1,
    borderRadius: 5,
    borderColor: 'white',
    borderBottomWidth: 0,
    shadowColor: 'black',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.24,
    shadowRadius: 2,
    elevation: 2,
    padding: 5,
    backgroundColor: 'white',
    justifyContent: 'flex-start',
    flexDirection: 'row'
  }
};

export { Card };
