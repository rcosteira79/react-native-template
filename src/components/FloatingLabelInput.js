import React from 'react';
import { View, TextInput, Animated } from 'react-native';

import EStyleSheet from 'react-native-extended-stylesheet';

class FloatingLabelInput extends React.Component {
  state = {
    isFocused: false
  };

  componentWillMount() {
    this.animatedIsFocused = new Animated.Value(this.props.value === '' ? 0 : 1);
  }

  componentDidUpdate() {
    Animated.timing(this.animatedIsFocused, {
      toValue: this.state.isFocused || this.props.value !== '' ? 1 : 0,
      duration: 200
    }).start();
  }

  handleFocus = () => this.setState({ isFocused: true });
  handleBlur = () => this.setState({ isFocused: false });

  handleAnimation(outputRange) {
    return this.animatedIsFocused.interpolate({
      inputRange: [0, 1],
      outputRange
    });
  }

  render() {
    const { label, unfocusedColor, focusedColor, refField, style, ...props } = this.props;
    const { viewStyle, labelStyle, textStyle } = styles;
    const labelStyleAnimations = {
      top: this.handleAnimation([35, 12]),
      fontSize: this.handleAnimation([14, 10]),
      color: this.handleAnimation([unfocusedColor, focusedColor])
    };

    const textStyleAnimations = {
      // Apparently, this can't be animated like the rest?
      borderBottomColor:
        this.state.isFocused || this.props.value !== '' ? focusedColor : unfocusedColor
    };

    return (
      <View style={[viewStyle, style]}>
        <Animated.Text style={[labelStyle, labelStyleAnimations]}>{label}</Animated.Text>
        <TextInput
          {...props}
          ref={refField}
          style={[textStyle, textStyleAnimations]}
          underlineColorAndroid={'transparent'}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
        />
      </View>
    );
  }
}

const styles = EStyleSheet.create({
  viewStyle: {
    alignSelf: 'stretch',
    paddingTop: 18
  },
  labelStyle: {
    position: 'absolute',
    height: 40,
    fontSize: 14,
    fontFamily: 'Montserrat-Light',
    left: 0
  },
  textStyle: {
    height: 40,
    fontSize: 14,
    fontFamily: 'Montserrat-Light',
    color: '$colorGrey',
    borderBottomWidth: 1
  }
});

export { FloatingLabelInput };
