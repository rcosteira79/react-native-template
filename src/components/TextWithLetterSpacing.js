import React from 'react';
import { View } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

import { Letter } from './Letter';

const spacingForLetterIndex = (letters, index, spacing) =>
  letters.length - 1 === index ? 0 : spacing;

export const TextWithLetterSpacing = props => {
  const { children, spacing, viewStyle, textStyle } = props;
  const letters = children.split('');

  return (
    <View style={[styles.container, viewStyle]}>
      {letters.map((letter, index) => (
        <Letter
          key={index}
          spacing={spacingForLetterIndex(letters, index, spacing)}
          textStyle={textStyle}
        >
          {letter}
        </Letter>
      ))}
    </View>
  );
};

const styles = EStyleSheet.create({
  container: {
    flexDirection: 'row'
  }
});
