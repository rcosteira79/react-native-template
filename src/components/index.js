export * from './FloatingLabelInput';
export * from './Card';
export * from './Button';
