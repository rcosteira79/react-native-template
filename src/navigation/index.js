import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addNavigationHelpers } from 'react-navigation';

import AppNavigator from './routes';

const AppWithNavigationState = ({ dispatch, navigation }) => {
  return (
    <AppNavigator
      navigation={addNavigationHelpers({
        dispatch,
        state: navigation
      })}
    />
  );
};

AppWithNavigationState.propTypes = {
  dispatch: PropTypes.func.isRequired,
  navigation: PropTypes.object.isRequired
};

const mapStateToProps = ({ navigation }) => ({
  navigation
});

export default connect(mapStateToProps)(AppWithNavigationState);
