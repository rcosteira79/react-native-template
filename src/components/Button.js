import React from 'react';
import EStyleSheet from 'react-native-extended-stylesheet';

import { TouchableOpacity } from 'react-native';
import { TextWithLetterSpacing } from './TextWithLetterSpacing';

const getButtonColor = enabled => {
  return EStyleSheet.value(
    enabled ? '$colorPrimary' : '$colorPrimaryDarkDisabled'
  );
};

const Button = ({ onPress, children, style, enabled }) => {
  const { buttonStyle, textViewStyle, textStyle } = styles;

  return (
    <TouchableOpacity
      style={[buttonStyle, style, { backgroundColor: getButtonColor(enabled) }]}
      disabled={!enabled}
      onPress={onPress}
    >
      <TextWithLetterSpacing
        spacing={2.4}
        viewStyle={textViewStyle}
        textStyle={textStyle}
      >
        {children}
      </TextWithLetterSpacing>
    </TouchableOpacity>
  );
};

const styles = EStyleSheet.create({
  textStyle: {
    color: 'white',
    fontSize: 14,
    fontFamily: 'Montserrat-Medium',
    letterSpacing: 2.4
  },
  textViewStyle: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  buttonStyle: {
    height: 50,
    borderRadius: 3,
    alignSelf: 'stretch'
  }
});

export { Button };
