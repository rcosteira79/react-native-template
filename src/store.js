import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import devTools from 'remote-redux-devtools';
import { localeReducer } from 'react-localize-redux';
import { persistStore, persistCombineReducers } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import { navigationReducer } from './navigation/reducer';
import { sessionReducer } from './services/session/reducer';
import { loginReducer } from './scenes/Login/reducer';
import { settingsReducer } from './scenes/Main/Settings/reducer';
import { GLOBAL } from './services/helpers/index';

const config = {
  key: 'root',
  storage,
  blacklist: ['navigation', 'locale']
};

const reducers = persistCombineReducers(config, {
  navigation: navigationReducer,
  session: sessionReducer,
  locale: localeReducer,
  login: loginReducer,
  settings: settingsReducer
});

const enhancer = compose(applyMiddleware(thunk), devTools({ realtime: GLOBAL.DEV }));

export const LOGOUT = 'LOGOUT';

export const store = createStore(reducers, enhancer);
export const persistor = persistStore(store);
