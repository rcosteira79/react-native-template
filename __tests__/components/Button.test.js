import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Button } from '../../src/components';

Enzyme.configure({ adapter: new Adapter() });

describe('Button component', () => {
  it('Is rendered enabled', () => {
    const tree = renderer.create(<Button enabled>Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Is rendered disabled', () => {
    const tree = renderer.create(<Button enabled={false}>Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Is clickable', () => {
    const handleClickMock = jest.fn();
    const buttonComponent = shallow(
      <Button enabled onPress={handleClickMock}>
        Text
      </Button>
    ).find('TouchableOpacity');

    buttonComponent.simulate('press');

    expect(handleClickMock).toBeCalled();
  });
});
