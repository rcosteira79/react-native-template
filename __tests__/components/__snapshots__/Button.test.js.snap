// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Is rendered disabled 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#16ADC164",
      "borderRadius": 3,
      "height": 50,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
            "letterSpacing": 2.4,
          },
          Object {
            "paddingRight": 2.4,
          },
        ]
      }
    >
      T
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
            "letterSpacing": 2.4,
          },
          Object {
            "paddingRight": 2.4,
          },
        ]
      }
    >
      e
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
            "letterSpacing": 2.4,
          },
          Object {
            "paddingRight": 2.4,
          },
        ]
      }
    >
      x
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
            "letterSpacing": 2.4,
          },
          Object {
            "paddingRight": 0,
          },
        ]
      }
    >
      t
    </Text>
  </View>
</View>
`;

exports[`Button component Is rendered enabled 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#55C5CE",
      "borderRadius": 3,
      "height": 50,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
            "letterSpacing": 2.4,
          },
          Object {
            "paddingRight": 2.4,
          },
        ]
      }
    >
      T
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
            "letterSpacing": 2.4,
          },
          Object {
            "paddingRight": 2.4,
          },
        ]
      }
    >
      e
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
            "letterSpacing": 2.4,
          },
          Object {
            "paddingRight": 2.4,
          },
        ]
      }
    >
      x
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 14,
            "letterSpacing": 2.4,
          },
          Object {
            "paddingRight": 0,
          },
        ]
      }
    >
      t
    </Text>
  </View>
</View>
`;
